examples/jsm/interactive/HTMLMesh.js

```diff
@@ -266,27 +284,164 @@ function html2canvas( element ) {
 
 			style = window.getComputedStyle( element );
 
+			// Get the border of the element used for fill and border
+
+			buildRectPath( x, y, width, height, parseFloat( style.borderRadius ) );
+
 			const backgroundColor = style.backgroundColor;
 
 			if ( backgroundColor !== 'transparent' && backgroundColor !== 'rgba(0, 0, 0, 0)' ) {
 
 				context.fillStyle = backgroundColor;
-				context.fillRect( x, y, width, height );
+				context.fill();
 
 			}
 
-			drawBorder( style, 'borderTop', x, y, width, 0 );
-			drawBorder( style, 'borderLeft', x, y, 0, height );
-			drawBorder( style, 'borderBottom', x, y + height, width, 0 );
-			drawBorder( style, 'borderRight', x + width, y, 0, height );
+			// If all the borders match then stroke the round rectangle
+
+			const borders = [ 'borderTop', 'borderLeft', 'borderBottom', 'borderRight' ];
+
+			let match = true;
+			let prevBorder = null;
+
+			for ( const border of borders ) {
+
+				if ( prevBorder !== null ) {
+
+					match = ( style[ border + 'Width' ] === style[ prevBorder + 'Width' ] ) &&
+					( style[ border + 'Color' ] === style[ prevBorder + 'Color' ] ) &&
+					( style[ border + 'Style' ] === style[ prevBorder + 'Style' ] );
+
+				}
+
+				if ( match === false ) break;
+
+				prevBorder = border;
+
+			}
+
+			if ( match === true ) {
+
+				// They all match so stroke the rectangle from before allows for border-radius
+
+				const width = parseFloat( style.borderTopWidth );
+
+				if ( style.borderTopWidth !== '0px' && style.borderTopStyle !== 'none' && style.borderTopColor !== 'transparent' && style.borderTopColor !== 'rgba(0, 0, 0, 0)' ) {
+
+					context.strokeStyle = style.borderTopColor;
+					context.lineWidth = width;
+					context.stroke();
+
+				}
+
+			} else {
+
+				// Otherwise draw individual borders
+
+				drawBorder( style, 'borderTop', x, y, width, 0 );
+				drawBorder( style, 'borderLeft', x, y, 0, height );
+				drawBorder( style, 'borderBottom', x, y + height, width, 0 );
+				drawBorder( style, 'borderRight', x + width, y, 0, height );
+
+			}
+
+			if ( element instanceof HTMLInputElement ) {
+
+				let accentColor = style.accentColor;
+
+				if ( accentColor === undefined || accentColor === 'auto' ) accentColor = style.color;
 
-			if ( element.type === 'color' || element.type === 'text' || element.type === 'number' ) {
+				color.set( accentColor );
 
-				clipper.add( { x: x, y: y, width: width, height: height } );
+				const luminance = Math.sqrt( 0.299 * ( color.r ** 2 ) + 0.587 * ( color.g ** 2 ) + 0.114 * ( color.b ** 2 ) );
+				const accentTextColor = luminance < 0.5 ? 'white' : '#111111';
```

Comment:
@AdaRoseCannon are these luminance coefficients chosen to match a particular (web?) specification, or are they more subjective and OK to change?

Sorry for bumping an old PR! I am considering how/if to update this in https://github.com/mrdoob/three.js/pull/28880, for consistency in color management across the library. These seem to be luminance coefficients for Rec. 601, rather than Rec. 709 / sRGB, but I just want to check if there's some additional reason here. Thanks!

Reply:
This was just for toggling the text color light/dark mode. It feels like a lifetime ago now but iirc I picked it because it closely matched Chrome's behaviour at the time so that what you see on the desktop closely matches the WebXR experience. Because it is just a binary toggle it's probably fine to change it however you need and it will probably work okay.

Reply:
Ok, that makes sense to me — thank you!